### ISSUE TITLE
LMDE 6] [Beta 1] Bluetooth keyboard does not automatically connect [Apple Magic A1843 Keyboard]

### ISSUE THREAD

**Issue by ghost** [linuxmint/lmde6-beta#20]
**Issue:** Bluetooth keyboard does not automatically connect after reboot and cannot be used to sign in at greeter.

**Frequency of issue:** Always

**Device info:**  

Apple Magic A1843 Keyboard

D4:DC:CD:7F:F7:22
public
Pac Edge’s Keyboard
Pac Edge’s Keyboard
0x002540
0x0000
input-keyboard
yes
yes
no
no
-58 dBm (0x-3a)
yes
00001124-0000-1000-8000-00805f9b34fb Human Interface Device Service (HID)
00001200-0000-1000-8000-00805f9b34fb PnP Information
bluetooth:v004Cp026Cd0160
/org/bluez/hci0
{76: [3, 0]}

**Comment by ghost** [comment]
The "auto connect" option, seen when right-clicking on the device listed, reads "Audio and input profiles." I have had that checked and unchecked with the same outcome.

**Comment by ghost** [comment]
System reports, System Information


System:
  Kernel: 6.1.0-12-amd64 arch: x86_64 bits: 64 compiler: gcc v: 12.2.0 Desktop: Cinnamon v: 5.8.4
    tk: GTK v: 3.24.37 wm: muffin dm: LightDM Distro: LMDE 6 Faye base: Debian 12.1 bookworm

Machine:
  Type: Desktop System: Dell product: Precision T3600 v: 01 serial: <superuser required> Chassis:
    type: 7 serial: <superuser required>
  Mobo: Dell model: 08HPGT v: A01 serial: <superuser required> UEFI: Dell v: A18 date: 06/25/2019

Battery:
  Device-1: hid-d4:dc:cd:7f:f7:22-battery model: Pac Edge’s Keyboard serial: N/A charge: N/A
    status: discharging

CPU:
  Info: 6-core model: Intel Xeon E5-1650 0 bits: 64 type: MT MCP arch: Sandy Bridge rev: 7 cache:
    L1: 384 KiB L2: 1.5 MiB L3: 12 MiB
  Speed (MHz): avg: 1662 high: 3800 min/max: 1200/3800 cores: 1: 3800 2: 1884 3: 1995 4: 1568
    5: 1200 6: 1200 7: 1371 8: 1341 9: 1200 10: 1200 11: 1200 12: 1995 bogomips: 76614
  Flags: avx ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx

Graphics:
  Device-1: AMD Navi 24 [Radeon RX 6400/6500 XT/6500M] vendor: Micro-Star MSI driver: amdgpu
    v: kernel arch: RDNA-2 pcie: speed: 16 GT/s lanes: 16 ports: active: HDMI-A-1 empty: DP-1
    bus-ID: 05:00.0 chip-ID: 1002:743f
  Display: x11 server: X.Org v: 1.21.1.7 with: Xwayland v: 22.1.9 driver: X: loaded: amdgpu
    unloaded: fbdev,modesetting,radeon,vesa dri: radeonsi gpu: amdgpu display-ID: :0 screens: 1
  Screen-1: 0 s-res: 3840x2160 s-dpi: 96
  Monitor-1: HDMI-A-1 mapped: HDMI-A-0 model: Toshiba TOSHIBA-TV res: 3840x2160 dpi: 61
    diag: 1836mm (72.3")
  API: OpenGL v: 4.6 Mesa 22.3.6 renderer: AMD Radeon RX 6400 (navi24 LLVM 15.0.6 DRM 3.49
    6.1.0-12-amd64) direct-render: Yes

Audio:
  Device-1: AMD Navi 21/23 HDMI/DP Audio driver: snd_hda_intel v: kernel pcie: speed: 16 GT/s
    lanes: 16 bus-ID: 05:00.1 chip-ID: 1002:ab28
  API: ALSA v: k6.1.0-12-amd64 status: kernel-api
  Server-1: PipeWire v: 0.3.65 status: active with: 1: pipewire-pulse status: active
    2: wireplumber status: active 3: pipewire-alsa type: plugin

Network:
  Device-1: Intel Wi-Fi 6 AX200 driver: iwlwifi v: kernel pcie: speed: 5 GT/s lanes: 1
    bus-ID: 0a:00.0 chip-ID: 8086:2723
  IF: wlp10s0 state: up mac: <filter>

Bluetooth:
  Device-1: Intel AX200 Bluetooth type: USB driver: btusb v: 0.8 bus-ID: 1-1.1:3 chip-ID: 8087:0029
  Report: hciconfig ID: hci0 rfk-id: 0 state: up address: <filter> bt-v: 3.0 lmp-v: 5.2
    sub-v: 200f

Drives:
  Local Storage: total: 670.7 GiB used: 21.36 GiB (3.2%)
  ID-1: /dev/sda vendor: Kingston model: SA400S37480G size: 447.13 GiB speed: 6.0 Gb/s
    serial: <filter>
  ID-2: /dev/sdb vendor: Kingston model: SA400S37240G size: 223.57 GiB speed: 6.0 Gb/s
    serial: <filter>

Partition:
  ID-1: / size: 445.13 GiB used: 8.66 GiB (1.9%) fs: btrfs dev: /dev/sda2
  ID-2: /boot/efi size: 2 GiB used: 5.8 MiB (0.3%) fs: vfat dev: /dev/sda1
  ID-3: /home size: 223.57 GiB used: 12.69 GiB (5.7%) fs: btrfs dev: /dev/sdb1

Swap:
  Alert: No swap data was found.

USB:
  Hub-1: 1-0:1 info: Full speed or root hub ports: 2 rev: 2.0 speed: 480 Mb/s chip-ID: 1d6b:0002
  Hub-2: 1-1:2 info: Intel Integrated Rate Matching Hub ports: 6 rev: 2.0 speed: 480 Mb/s
    chip-ID: 8087:0024
  Device-1: 1-1.1:3 info: Intel AX200 Bluetooth type: Bluetooth driver: btusb rev: 2.0
    speed: 12 Mb/s chip-ID: 8087:0029
  Hub-3: 2-0:1 info: Hi-speed hub with single TT ports: 2 rev: 2.0 speed: 480 Mb/s
    chip-ID: 1d6b:0002
  Hub-4: 2-1:2 info: VIA Labs USB2.0 Hub ports: 4 rev: 2.1 speed: 480 Mb/s chip-ID: 2109:2817
  Hub-5: 2-1.4:4 info: VIA Labs USB2.0 Hub ports: 4 rev: 2.1 speed: 480 Mb/s chip-ID: 2109:2817
  Hub-6: 2-1.4.4:5 info: VIA Labs USB2.0 Hub ports: 4 rev: 2.1 speed: 480 Mb/s chip-ID: 2109:2817
  Device-1: 2-2:3 info: SHARKOON GmbH [Mediatrack Edge Mini Keyboard] type: Keyboard,Mouse
    driver: hid-generic,usbhid rev: 1.1 speed: 12 Mb/s chip-ID: 1ea7:0066
  Hub-7: 3-0:1 info: Super-speed hub ports: 2 rev: 3.0 speed: 5 Gb/s chip-ID: 1d6b:0003
  Hub-8: 3-1:2 info: VIA Labs USB3.0 Hub ports: 4 rev: 3.1 speed: 5 Gb/s chip-ID: 2109:0817
  Hub-9: 3-1.4:3 info: VIA Labs USB3.0 Hub ports: 4 rev: 3.1 speed: 5 Gb/s chip-ID: 2109:0817
  Hub-10: 3-1.4.4:4 info: VIA Labs USB3.0 Hub ports: 4 rev: 3.1 speed: 5 Gb/s chip-ID: 2109:0817
  Hub-11: 4-0:1 info: Full speed or root hub ports: 2 rev: 2.0 speed: 480 Mb/s chip-ID: 1d6b:0002
  Hub-12: 4-1:2 info: Intel Integrated Rate Matching Hub ports: 8 rev: 2.0 speed: 480 Mb/s
    chip-ID: 8087:0024
  Device-1: 4-1.2:3 info: Chicony KU-0833 Keyboard type: Keyboard,HID driver: hid-generic,usbhid
    rev: 2.0 speed: 1.5 Mb/s chip-ID: 04f2:0833
  Device-2: 4-1.8:4 info: Brother Industries MFC-L2740DW type: Printer driver: usblp rev: 2.0
    speed: 480 Mb/s chip-ID: 04f9:0320

Sensors:
  System Temperatures: cpu: 57.0 C mobo: 26.0 C sodimm: SODIMM C gpu: amdgpu temp: 46.0 C
    mem: 44.0 C
  Fan Speeds (RPM): cpu: 951 gpu: amdgpu fan: 0

Repos:
  Packages: pm: dpkg pkgs: 2138
  No active apt repos in: /etc/apt/sources.list
  Active apt repos in: /etc/apt/sources.list.d/official-package-repositories.list
    1: deb http: //packages.linuxmint.com faye main upstream import backport
    2: deb https: //deb.debian.org/debian bookworm main contrib non-free non-free-firmware
    3: deb https: //deb.debian.org/debian bookworm-updates main contrib non-free non-free-firmware
    4: deb http: //security.debian.org bookworm-security main contrib non-free non-free-firmware
    5: deb https: //deb.debian.org/debian bookworm-backports main contrib non-free non-free-firmware

Info:
  Processes: 350 Uptime: 8h 29m Memory: 31.3 GiB used: 3.95 GiB (12.6%) Init: systemd v: 252
  target: graphical (5) default: graphical Compilers: gcc: 12.2.0 alt: 12 Client: Unknown
  python3.11 client inxi: 3.3.26

**Comment by ghost** [comment]
**Useful observation:**

Normally, both GDM and SDDM have the option to manually click and connect to Bluetooth, should automatically connect fail. This is a useful feature found in both (SDDM and GDM). In LMDE 6, while the button is there (if you install GDM or SDDM), it does not work.

Nevertheless, this seems to track as to why my Bluetooth keyboard was not automatically connecting at the greeter, preventing me from logging in.

Simply put, it cannot connect, because it seems the whole function is currently broken. Because even when installing GDM or SDDM, which supports the option, Mint will not allow connection until you're signed in.

**Comment by clefebvre** [comment]
Can you reproduce the issue with Debian 12 Cinnamon? I can't think of any reason why this would be Mint specific..

Regarding the feature in SDDM/GDM, if this can be supported by LightDM I'd be happy for us to consider it as a feature request for Slick-Greeter.
